pasqal-cloud/pasqal_cloud/authentication.py

```diff
@@ -146,8 +151,33 @@ def _query_token(self) -> dict[str, Any]:
             username=self.username,
             password=self.password,
             audience=self.auth0.audience,
-            scope="openid profile email",
+            scope=AUTH0_TOKEN_PROVIDER_SCOPE,
             realm=self.auth0.realm,
-            grant_type="http://auth0.com/oauth/grant-type/password-realm",
+            grant_type=AUTH0_TOKEN_PROVIDER_GRANT_TYPE,
         )
         return validated_token
+
+
+class InsecureAuth0TokenProvider(Auth0TokenProvider):
+    """Token Provider for auth0 with SSL verification disabled
+
+    Should be only used in testing environments.
+    """
+
+    def _query_token(self) -> Any:
+        response = requests.post(
+            "https://{}/oauth/token".format(self.auth0.domain),
+            data={
+                "client_id": self.auth0.public_client_id,
+                "username": self.username,
+                "password": self.password,
+                "realm": self.auth0.realm,
+                "client_secret": "",
+                "scope": AUTH0_TOKEN_PROVIDER_SCOPE,
+                "audience": self.auth0.audience,
+                "grant_type": AUTH0_TOKEN_PROVIDER_GRANT_TYPE,
+            },
+            verify=False,
+        )
+        response.raise_for_status()
+        return response.json()
```

Comment:
We could modify the `_query_token` method above to use this raw requests if the env skip ssl verify is defined. From user point of view, it will be simpler, you only have to set the env var once and everything is done for you.

Considering the short timeframe we have, I leave it up to you. It will be a beta anyway.

Reply:
Yeah I considered this; before doing this we would need to refine a bit the errors. The error messages are currently more detailed with the auth0 client than with the plain request. I want to release this ASAP so they can test and we can refactor down the road.